Tests/Sources/VLCLibraryTest.swift

```diff
@@ -0,0 +1,119 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+import XCTest
+
+class VLCLibraryTest: XCTestCase {
+    
+    let paramKey = "VLCParams"
+    
+    override func setUp() {
+        super.setUp()
+        UserDefaults.standard.removeObject(forKey: paramKey)
+    }
+
+    func testSharedLibrary() {
+        XCTAssertNotNil(VLCLibrary.shared())
+        XCTAssertNotNil(VLCLibrary.shared().instance)
+    }
+    
+    func testInitWithOptions() throws {
+        let tests = [
+            ["--no-video-title-show", "--verbose=4"],
+            []
+        ]
+        
+        for option in tests {
+            let library = try XCTAssertNotNilAndUnwrap(VLCLibrary(options: option))
+            XCTAssertNotNil(library.instance)
+            assertDefaultParameters()
+        }
+    }
+    
+    func testInit() {
+        let library = VLCLibrary()
+        XCTAssertNotNil(library.instance)
+        assertDefaultParameters()
+    }
+    
+    func testDebugLoggingLevel() throws {
+        let library = try XCTAssertNotNilAndUnwrap(VLCLibrary.shared())
+        XCTAssertEqual(library.debugLoggingLevel, 0)
+        
+        let tests: [(input: Int32, expected: Int32)] = [
+            (0, 0),
+            (3, 3),
+            (4, 4),
+            (100, 0),
+            (-10, 0)
+        ]
+        
+        for (input, expected) in tests {
+            library.debugLoggingLevel = input
+            XCTAssertEqual(library.debugLoggingLevel, expected)
+        }
+    }
+    
+    func testDebugLogging() throws {
+        let library = try XCTAssertNotNilAndUnwrap(VLCLibrary.shared())
+        XCTAssertFalse(library.debugLogging)
+        
+        library.debugLogging = true
+        XCTAssertTrue(library.debugLogging)
+        
+        library.debugLogging = false
+        XCTAssertFalse(library.debugLogging)
+    }
+    
+    func testLibraryDescription() throws {
+        let library = try XCTAssertNotNilAndUnwrap(VLCLibrary.shared())
+        
+        XCTAssert(library.version.count > 1)
+        XCTAssert(library.compiler.count > 1)
+        XCTAssert(library.changeset.count > 1)
+    }
+}
+
+extension VLCLibraryTest {
+    func assertDefaultParameters() {
+        let expected = [
+            "--play-and-pause",
+            "--no-color",
+            "--no-video-title-show",
+            "--verbose=4",
+            "--no-sout-keep",
+            "--vout=macosx",
+            "--text-renderer=freetype",
+            "--extraintf=macosx_dialog_provider",
+            "--audio-resampler=soxr"
+        ]
+        
+        let defaultParams = UserDefaults.standard.object(forKey: paramKey)
+        
+        #if os(macOS)
+            XCTAssertEqual(defaultParams as? [String], expected)
```

Comment:
When is defaultParams ever nil? Does it make sense to make this also not optional?

Reply:
`defaultParams` is set in `VLCLibrary` during an `init`. But the user could override it by using the same key by mistake since the parameters are stored in `UserDefaults.standard`

Reply:
I'm still confused a bit by this test :D  so I overread the nil two lines down for iOS and I thought this test makes sure that the defaultParams are always not nil and set?

Reply:
`VLCLibrary.m` **DOES NOT** set the default parameters in the UserDefaults `#if TARGET_OS_IPHONE`. It saves the default params in UserDefaults otherwise. https://github.com/videolan/vlckit/blob/ba9d2cd7e5099c480df0e7e3a1376508f9243f1b/Sources/VLCLibrary.m#L101